libphobos/libdruntime/core/cpuid.d

```diff
@@ -934,6 +970,42 @@ bool hasCPUID()
     }
 }
 
+version (GNU)
+{
+    nothrow @nogc
+    void gnuCpuid(uint eax, out uint a, out uint b, out uint c, out uint d)
+    {
+        gnuCpuid(eax, 0, a, b, c, d);
+    }
+
+    nothrow @nogc
+    void gnuCpuid(uint eax, uint ecx, out uint a, out uint b, out uint c, out uint d)
+    {
+        // On x86 the PIC register 'ebx' must not be overwritten. GCC <5 will error out otherwise.
+        version (X86) version (D_PIC)
+            enum SaveEbx;
+        
+        static if (is(SaveEbx))
+        {
+            asm nothrow @nogc
+            {
+                "push {%%}ebx; cpuid; mov {%%ebx, %1|%1, ebx}; pop {%%}ebx"
```

Comment:
Same as %1, just explicitly using the 32 bit registers.
https://gcc.gnu.org/onlinedocs/gcc/Extended-Asm.html#x86Operandmodifiers